server/game/gameSystems/SelectCardSystem.ts

```diff
@@ -164,4 +164,11 @@ export class SelectCardSystem<TContext extends AbilityContext = AbilityContext>
         const properties = this.generatePropertiesFromContext(context, additionalProperties);
         return properties.checkTarget && properties.player !== RelativePlayer.Opponent;
     }
+
+    private selectionIsOptional(properties, context): boolean {
+        if (properties.innerSystem.isOptional(context)) {
+            return true;
+        }
+        return CardTargetResolver.allZonesAreHidden(properties.locationFilter);
```

Comment:
Want to confirm that this behavior is correct in all cases - what about a situation that's like "the opponent reveals their hand, then you choose a card to discard?" Would this logic cause it to become optional when it shouldn't?

My gut feeling is that the selector can't have enough context to determine when the hidden-ness rule is relevant so it will have to be a property or something, but I could be wrong

Reply:
That is a good point(choosing from an opponent's otherwise hidden zone). Do we have any cards implemented yet that do that?
Speculatively, I think the easy thing to do to account for your example would be to add `controller` and `selectingPlayer` arguments to `allZonesAreHidden()`. Then we can say: "if the selecting player is selecting from their opponent's hand, then it's not really a hidden zone".

Reply:
We don't have anything implemented yet, the discard systems aren't in place yet unfortunately, I think your idea should work though. We can put in the logic for now and maybe just leave a TODO comment that not all cases have been tested.